packages/logic-utils/src/utils/documentation.ts

```diff
@@ -0,0 +1,191 @@
+import {
+  UI5DeprecatedInfo,
+  UI5SemanticModel,
+} from "@ui5-language-assistant/semantic-model-types";
+import { ReplaceFunction, forEach, unescape } from "lodash";
+
+const MAX_SNIPPET_LENGTH = 500;
+
+export function getDeprecationPlainTextSnippet(
+  title: string | undefined,
+  deprecatedInfo: UI5DeprecatedInfo,
+  model: UI5SemanticModel
+): string {
+  let text = deprecatedInfo.text;
+  if (text !== undefined) {
+    text = convertJSDocToPlainText(text, model);
+    // Only take the first sentence.
+    // A sentence is defined by a dot followed by a whitespace character.
+    // TODO Should we allow the result to have more than 1 line? (If so, add "s" flag to the regex)
+    const matches = text.match(/^((?:.*?)\.)\s/);
+    if (matches !== null) {
+      text = matches[1];
+    }
+    // Limit the length of the snippet
+    if (text.length > MAX_SNIPPET_LENGTH) {
+      text = text.substring(0, MAX_SNIPPET_LENGTH) + "...";
+    }
+  }
+
+  const doc = getDeprecationMessage({
+    title,
+    since: deprecatedInfo.since,
+    text,
+  });
+  return doc;
+}
+
+export function getDeprecationMessage({
+  title,
+  since,
+  text,
+}: {
+  title?: string;
+  since: string | undefined;
+  text: string | undefined;
+}): string {
+  let contents = title ?? "Deprecated";
+  contents += since ? ` since version ${since}` : "";
+  contents += ".";
+  contents += text ? ` ${text}` : "";
+  return contents;
+}
+
+function convertJSDocToPlainText(
+  jsdocDescription: string,
+  model: UI5SemanticModel
+): string {
+  return convertJSDoc(jsdocDescription, "plaintext", model);
+}
+
+export function convertJSDocToMarkdown(
+  jsdocDescription: string,
+  model: UI5SemanticModel
+): string {
+  return convertJSDoc(jsdocDescription, "markdown", model);
+}
+
+type ConvertTarget = "markdown" | "plaintext";
+function convertJSDoc(
+  jsdoc: string,
+  target: ConvertTarget,
+  model: UI5SemanticModel
+): string {
+  // The replacements are applied in order
+  const tagMatcherToReplacement: {
```

Comment:
Maybe subjective, but is there a value to defining the model outside the function?

- I tend to prefer separation of model and logic.
- In this case the model is semi complex, would all these regexps be initialized everytime the `convertJSDoc` function is called?

Reply:
I'll move it outside for efficiency but it's part of the function logic, at least as long as we don't allow to send anything else (like a custom type instead of `plaintext` and `markdown`) to the function (and if we do then we should define a proper interface with a method for each replacement, but it seems like overkill for the 2 use cases we have now).

Reply:
Moved